src/Symfony/Component/Mailer/Bridge/Amazon/Http/SesTransport.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Mailer\Bridge\Amazon\Http;
+
+use Psr\Log\LoggerInterface;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use Symfony\Component\HttpClient\HttpClient;
+use Symfony\Component\Mailer\Exception\TransportException;
+use Symfony\Component\Mailer\SentMessage;
+use Symfony\Component\Mailer\Transport\AbstractTransport;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * @experimental in 4.3
+ */
+class SesTransport extends AbstractTransport
+{
+    private $endpoint = 'https://email.%region%.amazonaws.com';
+    private $client;
+    private $accessKey;
+    private $secretKey;
+    private $region;
+
+    /**
+     * @param string $region Amazon SES region (currently one of us-east-1, us-west-2, or eu-west-1)
+     */
+    public function __construct(string $accessKey, string $secretKey, string $region = null, HttpClientInterface $client = null, EventDispatcherInterface $dispatcher = null, LoggerInterface $logger = null)
+    {
+        $this->client = $client ?? HttpClient::create();
+        $this->accessKey = $accessKey;
+        $this->secretKey = $secretKey;
+        $this->region = $region ?: 'eu-west-1';
+
+        parent::__construct($dispatcher, $logger);
+    }
+
+    protected function doSend(SentMessage $message): void
+    {
+        $date = gmdate('D, d M Y H:i:s e');
+        $auth = sprintf('AWS3-HTTPS AWSAccessKeyId=%s,Algorithm=HmacSHA256,Signature=%s', $this->accessKey, $this->getSignature($date));
```

Comment:
shouldnt we share this logic somehow.. (duplicated above)

Reply:
Not sure if having an abstract class/trait would really be better. I will keep it this way for now.